android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java

```diff
@@ -53,64 +53,135 @@ of this software and associated documentation files (the "Software"), to deal
+import com.onesignal.inAppMessages.IInAppMessageWillDisplayEvent;
+import com.onesignal.inAppMessages.IInAppMessageDidDisplayEvent;
+import com.onesignal.inAppMessages.IInAppMessageWillDismissEvent;
+import com.onesignal.inAppMessages.IInAppMessageDidDismissEvent;
 import com.onesignal.notifications.INotification;
-import com.onesignal.notifications.INotificationClickHandler;
-import com.onesignal.notifications.INotificationClickResult;
-import com.onesignal.notifications.INotificationReceivedEvent;
-import com.onesignal.notifications.INotificationWillShowInForegroundHandler;
-import com.onesignal.notifications.IPermissionChangedHandler;
+import com.onesignal.notifications.INotificationClickListener;
+import com.onesignal.notifications.INotificationClickEvent;
+import com.onesignal.notifications.INotificationLifecycleListener;
+import com.onesignal.notifications.INotificationWillDisplayEvent;
+import com.onesignal.notifications.IPermissionObserver;
 import com.onesignal.user.subscriptions.IPushSubscription;
-import com.onesignal.user.subscriptions.ISubscription;
-import com.onesignal.user.subscriptions.ISubscriptionChangedHandler;
+import com.onesignal.user.subscriptions.IPushSubscriptionObserver;
+import com.onesignal.user.subscriptions.PushSubscriptionState;
+import com.onesignal.user.subscriptions.PushSubscriptionChangedState;
 import org.json.JSONException;
 import org.json.JSONObject;
 
 import java.util.HashMap;
 
 public class RNOneSignal extends ReactContextBaseJavaModule implements
-        ISubscriptionChangedHandler,
-        IPermissionChangedHandler,
+        IPushSubscriptionObserver,
+        IPermissionObserver,
         LifecycleEventListener,
-        INotificationWillShowInForegroundHandler{
+        INotificationLifecycleListener{
     private ReactApplicationContext mReactApplicationContext;
     private ReactContext mReactContext;
 
     private boolean oneSignalInitDone;
     private boolean hasSetPermissionObserver = false;
     private boolean hasSetPushSubscriptionObserver = false;
 
-    private HashMap<String, INotificationReceivedEvent> notificationReceivedEventCache;
-    private boolean hasSetNotificationWillShowInForegroundHandler = false;
+    private HashMap<String, INotificationWillDisplayEvent> notificationWillDisplayCache;
+    private HashMap<String, INotificationWillDisplayEvent> preventDefaultCache;
+
+    private boolean hasAddedNotificationForegroundListener = false;
+    private boolean hasAddedInAppMessageLifecycleListener = false;
+
+    private IInAppMessageClickListener rnInAppClickListener = new IInAppMessageClickListener() {
+        @Override
+        public void onClick(IInAppMessageClickEvent event) {
+            try {
+                IInAppMessageClickResult result = event.getResult();
+                sendEvent("OneSignal-inAppMessageClicked", RNUtils.convertHashMapToWritableMap(RNUtils.convertInAppMessageClickResultToMap(result)));
```

Comment:
Pass the whole InAppMessageClickEvent

Reply:
Addressed in [Update Android Notification and IAM listeners to send entire event](https://github.com/OneSignal/react-native-onesignal/pull/1532/commits/762ee2b05d3c4c6e272c7c4a3b994f0e00575b59)